ledger/error/generator/app/src/main/scala/com/daml/error/generator/app/ErrorCodeInventoryDocsGenApp.scala

```diff
@@ -0,0 +1,204 @@
+  ): Unit = {
+    insert(
+      remaining = errorCode.errorGroupPath.groupings,
+      path = Nil,
+      errorCode = errorCode,
+      getExplanation = getExplanation,
+    )
+  }
+
+  private def insert(
+      remaining: List[Grouping],
+      errorCode: ErrorCodeValue,
+      path: List[Grouping],
+      getExplanation: (List[Grouping]) => Option[String],
+  ): Unit = {
+
+    remaining match {
+      case Nil =>
+        assert(!errorCodes.contains(errorCode.code), s"Code: ${errorCode.code} is already present!")
+        errorCodes.put(errorCode.code, errorCode): Unit
+      case headGroup :: tail =>
+        val newPath = path :+ headGroup
+        if (!children.contains(headGroup)) {
+          children.put(
+            headGroup,
+            new ErrorGroupTree(
+              name = headGroup.docName,
+              explanation = getExplanation(newPath),
+            ),
+          )
+        }
+        children(headGroup).insert(
+          remaining = tail,
+          errorCode = errorCode,
+          path = newPath,
+          getExplanation,
+        )
+    }
+  }
+
+}
+
+object ErrorGroupTree {
+  def empty(): ErrorGroupTree = new ErrorGroupTree(
+    name = "",
+    explanation = None,
+  )
+
+  def collectErrorCodesAsReStructuredTextSubsections(root: ErrorGroupTree): List[String] = {
+
+    // in-order tree traversal
+    def iter(
+        tree: ErrorGroupTree,
+        path: List[String],
+        groupHierarchicalIndex: List[Int],
+    ): List[String] = {
+      val newPath = path :+ tree.name
+      val textBuffer: mutable.ArrayBuffer[String] = new ArrayBuffer[String]()
+
+      // Add group text
+      textBuffer.addOne(s"""${groupHierarchicalIndex.mkString(".")}. ${newPath.mkString(" / ")}
+           |-------------------------------------------------------------------------------------------------------------------
+           |
+           |${tree.explanation.getOrElse("")}
+           |""".stripMargin)
+      // Add error codes in this group
+      textBuffer.addAll(
+        tree
+          .sortedErrorCodes()
+          .map(handleErrorCode)
+      )
+      // Recurse to sub-groups
+      textBuffer.addAll(
+        tree
+          .sortedSubGroups()
+          .zipWithIndex
+          .flatMap { case (subGroup: ErrorGroupTree, index: Int) =>
+            iter(
+              subGroup,
+              newPath,
+              groupHierarchicalIndex = groupHierarchicalIndex :+ (index + 1),
+            )
+          }
+      )
+      textBuffer.toList
+    }
+
+    root
+      .sortedSubGroups()
+      .zipWithIndex
+      .flatMap { case (subGroup, i) =>
+        iter(subGroup, path = List(), groupHierarchicalIndex = List(i + 1))
+      }
+  }
+
+  private def handleErrorCode(e: ErrorCodeValue): String = {
+    val deprecationText = e.deprecationO.fold("")(d => s"""
+         |    **Depreciation**: ${d}
+         |    """.stripMargin)
+    s"""Error code: ${e.code}
```

Comment:
```suggestion
    s"""${e.code}
```
I think it's fine without the prefix

Reply:
Ok